indico/modules/events/contributions/controllers.py

```diff
@@ -175,3 +182,61 @@ class RHContributionPersonList(RHContributionPersonListMixin, RHManageContributi
     def _membership_filter(self):
         contribution_ids = {contrib.id for contrib in self.contribs}
         return Contribution.id.in_(contribution_ids)
+
+
+class RHContributionProtection(RHManageContributionBase):
+    """Manage contribution protection"""
+
+    def _process(self):
+        form = ContributionProtectionForm(obj=FormDefaults(**self._get_defaults()))
+        if form.validate_on_submit():
+            self._handle_contrib_managers(form.managers.data)
+            self._handle_protection_mode(form.protection_mode.data)
+            self._handle_access_control_list(form.acl.data)
+            self._handle_contrib_submitters(form.submitters.data)
+            return jsonify_data(flash=False)
+        protection_message = _render_contrib_protection_message(self.contrib)
+        return jsonify_template('events/contributions/management/contribution_protection.html', form=form,
+                                protection_message=protection_message)
+
+    def _get_defaults(self):
+        managers = {p.principal for p in self.contrib.acl_entries if p.full_access}
+        submitters = {p.principal for p in self.contrib.acl_entries if 'submit' in p.roles}
+        acl = {p.principal for p in self.contrib.acl_entries if p.read_access}
+        return {'managers': managers, 'submitters': submitters, 'protection_mode': self.contrib.protection_mode.name,
+                'acl': acl}
+
+    def _handle_contrib_managers(self, managers_data):
+        current_managers = {(acl.principal.name if acl.principal.is_group else acl.principal.id): acl.principal
```

Comment:
Actually, I think you can do something like this (make it more generic so you can use it for the acl too instead of having to duplicate code):

```python

current_managers = {x.principal for x in self.contrib.acl_entries if x.full_access}
for principal in managers_data - current_managers:
    self.contrib.update_principal(principal, full_access=True)
for principal in current_managers - managers_data:
    self.contrib.update_principal(principal, full_access=False)
```